src/main/java/frc/robot/commands/LevelRobot.java

```diff
@@ -0,0 +1,57 @@
+package frc.robot.commands;
+
+import com.ctre.phoenix.sensors.WPI_Pigeon2;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.ExampleSubsystem;
+import frc.robot.subsystems.Swerve;
+public class LevelRobot extends CommandBase{
+    private final Swerve drive;
+    private final PIDController pid = new PIDController(0.1, 0.001, 0.005);
+    private final PIDController headingPid = new PIDController(0.04, 0, 0.0025);
+    private double startingYaw;
+    
+    public LevelRobot(Swerve drivetrain){
+        drive = drivetrain;
+        addRequirements(drive);
+    }
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+        pid.setTolerance(0);
+        pid.setSetpoint(0);
+        startingYaw = drive.getYawAsDouble() % 360;
+        System.out.println("Command Init");
+    }
+
+    // Called every time the scheduler runs while the command is scheduled.
+    @Override
+    public void execute() {
+        double roll = drive.getRollAsDouble();
+        double yaw = drive.getYawAsDouble() % 360;
+        double deltaYaw = yaw-startingYaw;
+        double calculationMovement = pid.calculate(roll, 0);
+        double calculationHeading = headingPid.calculate(deltaYaw * -1, 0);
+        calculationMovement = MathUtil.clamp(calculationMovement, -0.45, 0.45);
+        calculationHeading = MathUtil.clamp(calculationHeading, -0.5, 0.5);
```

Comment:
Factor the clamp constants into something descriptive you can update easily later? Perhaps something like "leveling drive speed limit" and "leveling heading speed limit"?

Reply:
All the numbers in here are now a part of the constants file